### PR TITLE
New scenario for kafka broker scale up

### PR DESCRIPTION
**Pull request by anuj-apdev**
# Description

New scenario for brokers scale up.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

By running newly created molecule scenario and verify.

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/broker-scale-up/converge.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+- name: Add new brokers Nodes into in-memory inventory to scale-up
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: "Add new Broker Nodes"
```
Comment (utkarsh5474) [inline molecule/broker-scale-up/converge.yml:6]: A question for my understanding - If the host is present in the molecule file, will this module modify the host details?

File: molecule/broker-scale-up/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      - /sys/fs/cgroup:/sys/fs/cgroup:ro
+    privileged: true
+    published_ports:
+      - "9021:9021"
+    networks:
+      - name: confluent
+provisioner:
+  playbooks:
+    converge: converge.yml
+  inventory:
```
Comment (utkarsh5474) [inline molecule/broker-scale-up/molecule.yml:154]: I think we should enable sbc for this scenario with kafka_broker_custom_properties as that'll be closer to customer's use case. WDYT?

**Review by utkarsh5474 — COMMENTED** [review]
A few minor comments. Looks cool

File: molecule/broker-scale-up/converge.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      add_host:
+        hostname: kafka-broker4
+        ansible_connection: community.docker.docker
```
Comment (nsharma-git) [inline molecule/broker-scale-up/converge.yml:9]: Is this required. Is the top level driver name not sufficient here?

File: molecule/broker-scale-up/converge.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        groups: kafka_broker
+        tag: kafka_broker
```
Comment (nsharma-git) [inline molecule/broker-scale-up/converge.yml:11]: Please dont use tag in tests

File: molecule/broker-scale-up/converge.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    - name: "Add new Broker Nodes"
```
Comment (nsharma-git) [inline molecule/broker-scale-up/converge.yml:13]: better task name please. 
Add new broker node - {number}

File: molecule/broker-scale-up/converge.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      add_host:
+        hostname: kafka-broker5
+        ansible_connection: community.docker.docker
+        groups: kafka_broker
+        tag: kafka_broker
+
+    - name: "Print Test"
```
Comment (nsharma-git) [inline molecule/broker-scale-up/converge.yml:20]: Please remove the debug tasks. 
You can have similar assertions in the verify phase.

File: molecule/broker-scale-up/molecule.yml [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        kafka_connect_ssl_mutual_auth_enabled: false
+        kafka_connect_connectors:
+          - name: sample-connector-2
+            config:
+              connector.class: "org.apache.kafka.connect.tools.VerifiableSinkConnector"
+              tasks.max: "1"
+              file: "/etc/kafka/connect-distributed.properties"
+              topics: "test_topic"
+
+      # ksql clusters
```
Comment (nsharma-git) [inline molecule/broker-scale-up/molecule.yml:195]: I think most of these variables is not required here. Please get it cleaned

File: molecule/broker-scale-up/converge.yml (hunk 1 above)
Comment (anuj-apdev) [reply molecule/broker-scale-up/converge.yml:6]: Since host is there in molecule file, create phase will bring it's container up, but It's not part of kafka_broker cluster.

In converge phase, this command will add this as a new host (duplicates are fine in ansible inventory.) But more effectively, this `add_host` module will add this node to `kafka_broker` host group for in memory inventory.
Now next platform.all playbook call will consider this change.

File: molecule/broker-scale-up/verify.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- name: Verify - kafka_brokers
+  hosts: kafka_broker
+  gather_facts: false
+  tasks:
+    - name: Get Broker's URL
+      uri:
+        url: "https://kafka-broker3:8090/kafka/v3/clusters"
+        status_code: 200
+        validate_certs: false
+      register: broker_uri
```
Comment (nsharma-git) [inline molecule/broker-scale-up/verify.yml:11]: better name - broker_details / get_broker_response

File: molecule/broker-scale-up/molecule.yml (hunk 2 above)
Comment (anuj-apdev) [reply molecule/broker-scale-up/molecule.yml:154]: Totally makes sense. Adding it.

**Review by nsharma-git — CHANGES_REQUESTED** [review]

File: molecule/broker-scale-up/molecule.yml [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+      - name: confluent
+provisioner:
+  playbooks:
+    converge: converge.yml
+  inventory:
+    host_vars:
+      # ksql1 will not have an advertised listener, validating the defaulting logic
+      ksql2:
+        ksql_advertised_listener_hostname: ksql2.confluent
+      ksql3:
```
[the remark this answers is not in the document]
Comment (anuj-apdev) [reply molecule/broker-scale-up/molecule.yml:159]: That was a miss. I removed ksql3 (which was originally there) from inventory to save some memory and resources. However i missed to clean up it's properties, which is useless.
Thanks for pointing it out.

File: molecule/broker-scale-up/molecule.yml (hunk 7 above)
Comment (nsharma-git) [reply molecule/broker-scale-up/molecule.yml:193]: @anuj-apdev any update on this

File: molecule/broker-scale-up/molecule.yml (hunk 2 above)
Comment (anuj-apdev) [reply molecule/broker-scale-up/molecule.yml:154]: Enabled Line # 164

File: molecule/broker-scale-up/molecule.yml [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+    command: ""
+    volumes:
+      - /sys/fs/cgroup:/sys/fs/cgroup:ro
+    privileged: true
+    networks:
+      - name: confluent
+  - name: ksql2
+    hostname: ksql2.confluent
+    groups:
+      - ksql
```
Comment (nsharma-git) [inline molecule/broker-scale-up/molecule.yml:127]: Can we avoid multiple groups. We can remove all group occurrences

File: molecule/broker-scale-up/molecule.yml [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+              file: "/etc/kafka/connect-distributed.properties"
+              topics: "test_topic"
+
+      # ksql clusters
+      ks1:
+        ksql_service_id: ksql1_
+
+      control_center:
+        ksql_cluster_ansible_group_names:
+          - ks1
```
Comment (nsharma-git) [inline molecule/broker-scale-up/molecule.yml:188]: Is this relevant to this test?

File: molecule/broker-scale-up/molecule.yml [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+          - name: sample-connector-1
+            config:
+              connector.class: "org.apache.kafka.connect.tools.VerifiableSinkConnector"
+              tasks.max: "1"
+              file: "/etc/kafka/connect-distributed.properties"
+              topics: "test_topic"
+
+      # ksql clusters
+      ks1:
+        ksql_service_id: ksql1_
```
Comment (nsharma-git) [inline molecule/broker-scale-up/molecule.yml:184]: Please get rid of this

File: molecule/broker-scale-up/converge.yml [hunk 13]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Add new brokers Nodes into in-memory inventory to scale-up
```
Comment (nsharma-git) [inline molecule/broker-scale-up/converge.yml:2]: Can we add some validation to ensure these nodes are not part of the cluster.

**Review by nsharma-git — APPROVED** [review]
